adr/0012-bigbang-as-a-noun.md

```diff
@@ -0,0 +1,39 @@
+# 12. BigBang as a Noun
+
+Date: 2023-01-18
+
+## Status
+
+Accepted
+
+## Context
+
+One primary application component that end users of Zarf are deploying is [Big Bang](https://repo1.dso.mil/platform-one/big-bang/bigbang).  The installation of BigBang is complicated for several reason:
+
+- It requires Flux to be installed to deploy correctly do to the use of Flux CRDs. 
+- The [images](https://umbrella-bigbang-releases.s3-us-gov-west-1.amazonaws.com/umbrella/1.51.0/package-images.yaml) defined within BigBang are normally a superset of the images needed for any individual deployment.
+- All images that BigBang might need takes 10s of gigabytes of storage to include in a Zarf package.
+- The git repositories defined within BigBang are normally a superset of the git repositories needed for any individual deployment.
+- Injecting a `values.yaml` file into the [default deployment structure](https://repo1.dso.mil/big-bang/bigbang/-/blob/master/base/kustomization.yaml) is complicated.
+and the discovery of which images are needed is a function of the values that are provided to the BigBang chart
+
+
+## Decision
+
+Deployments of BigBang can be managed with a new `bigbang` noun in the zarf.yaml that manages the complexity of the deployment.  This capability will take the values provided to the big bang chart, template them during the package phase to identify which [BigBang packages](https://repo1.dso.mil/big-bang/bigbang/-/blob/master/docs/packages.md) are being configured in the Zarf package.  The code then includes only the git repositories and images needed for the configured packages, and does not include the git repositories and images for packages that would not be deployed.  
+
+
+ The `bigbang` section will provide the following configurations for managing a big bang deployment:
+
+- `version` - Identifies the particular version of Bigbang to deploy, which correspond to git tags in the provided `repo`.  See versions of BigBang [here](https://repo1.dso.mil/big-bang/bigbang/-/releases).  
+- `repo` - Identifies the git repository BigBang is hosted on.  Defaults to https://repo1.dso.mil/platform-one/big-bang/bigbang.git
+- `valuesFrom` - list of local files that get passed to the BigBang helm chart for deployment. 
```

Comment:
```suggestion
- `valuesFrom` - list of local files that get added to the zarf package at create time which will be applied to the Big Bang helm chart. Deploy time variables are able to be declared as part of said values files.
```

@runyontr does this track with your thinking as well?

Reply:
Yes, we should have deploy time variables able to be injected into those values.yaml files.  We'd need to make sure they get a value in package so we can pull the correct images/repos though.  @corang is that how its implemented currently?

Reply:
any further comment on this?